internal/internal.go

```diff
@@ -0,0 +1,10 @@
+package internal
+
+import (
+	"path"
+)
+
+// WithinNamespace generates a full path scoped within a namespace.
+func WithinNamespace(ns, name string) string {
+	return path.Join(ns, path.Clean("/"+name))
```

Comment:
Oh. I don't remember if we ensure `ns` to be sane in "drivers". And I'm not sure that it'll never change:

```
	ns := ""
	name := "/etc/passwd"
	fmt.Println(path.Join(ns, path.Clean("/"+name))) // "/etc/passwd"
```

I'd suggest to do smth like `ns ||= "."` here just to be safe.

Also, I'd vote to use `path.Join("/", name)` instead of string concatenation - looks more, IDK, idiomatic.

Reply:
not my code: https://golang.org/src/net/http/fs.go#L74

Reply:
:scream: